lasagne/layers/special.py

```diff
@@ -530,3 +534,102 @@ def prelu(layer, **kwargs):
     if nonlinearity is not None:
         layer.nonlinearity = nonlinearities.identity
     return ParametricRectifierLayer(layer, **kwargs)
+
+
+class RandomizedRectifierLayer(Layer):
+    """
+    lasagne.layers.RandomizedRectifierLayer(incoming,
+    lower=init.Constant(0.3), upper=init.Constant(0.8),
+    shared_axes='auto', **kwargs)
+
+    The randomized leaky rectifier was first proposed and used in the Kaggle
+    NDSB Competition, and later evaluated in [1]_. Compared to the standard
+    leaky rectifier :func:`leaky_rectify`, it has a randomly sampled gradient
+    for negative input during training, and a fixed gradient during evaluation.
+
+    Equation for the randomized rectifier linear unit during training:
+    :math:`\\varphi(x) = \\max((\\sim U(lower, upper)) \\cdot x, x)`
+
+    During evaluation, the factor is fixed to the arithmetic mean of `lower`
+    and `upper`.
+
+    Parameters
+    ----------
+    incoming : a :class:`Layer` instance or a tuple
+        The layer feeding into this layer, or the expected input shape
+
+    lower : Theano shared variable, expression, or constant
+        The lower bound for the randomly chosen leaky gradient.
+
+    upper : Theano shared variable, expression, or constant
+        The upper bound for the randomly chosen leaky gradient.
+
+    **kwargs
+        Any additional keyword arguments are passed to the `Layer` superclass.
+
+     References
+    ----------
+    .. [1] Bing Xu, Naiyan Wang et al. (2015):
+       Empirical Evaluation of Rectified Activations in Convolutional Network,
+       http://arxiv.org/abs/1505.00853
+    """
+    def __init__(self, incoming, lower=init.Constant(0.3),
+                 upper=init.Constant(0.8), shared_axes='auto',
```

Comment:
`lower` and `upper` should just be ordinary Python scalars, just like the dropout rate in the `DropoutLayer`. If you change that, you don't need to include the function signature in the docstring any more (it's just needed if the default parameters contain some classes that don't have nice string representations).